### PR TITLE
adjust builtIn entities for microservices

### PR DESCRIPTION
**Pull request by mshima**
Fixes https://github.com/jhipster/generator-jhipster/issues/25550

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mraible** [comment]
If I try this change with `jhipster jdl reactive-mf.jdl --monorepository --workspaces --client-framework angular`, the blog app fails to compile.

```
> Task :compileJava FAILED
warning: unknown enum constant When.MAYBE
  reason: class file for javax.annotation.meta.When not found
/Users/mraible/Downloads/micro-angular/blog/src/main/java/com/okta/developer/blog/domain/User.java:60: error: cannot find symbol
    private Set<Authority> authorities = new HashSet<>();
                ^
  symbol:   class Authority
  location: class User
/Users/mraible/Downloads/micro-angular/blog/src/main/java/com/okta/developer/blog/domain/User.java:127: error: cannot find symbol
    public Set<Authority> getAuthorities() {
               ^
  symbol:   class Authority
  location: class User
/Users/mraible/Downloads/micro-angular/blog/src/main/java/com/okta/developer/blog/domain/User.java:131: error: cannot find symbol
    public void setAuthorities(Set<Authority> authorities) {
                                   ^
  symbol:   class Authority
  location: class User
/Users/mraible/Downloads/micro-angular/blog/src/main/java/com/okta/developer/blog/service/UserService.java:4: error: cannot find symbol
import com.okta.developer.blog.domain.Authority;
                                     ^
  symbol:   class Authority
  location: package com.okta.developer.blog.domain
/Users/mraible/Downloads/micro-angular/blog/src/main/java/com/okta/developer/blog/service/UserService.java:6: error: cannot find symbol
import com.okta.developer.blog.repository.AuthorityRepository;
                                         ^
  symbol:   class AuthorityRepository
  location: package com.okta.developer.blog.repository
/Users/mraible/Downloads/micro-angular/blog/src/main/java/com/okta/developer/blog/service/UserService.java:35: error: cannot find symbol
    private final AuthorityRepository authorityRepository;
                  ^
  symbol:   class AuthorityRepository
  location: class UserService
/Users/mraible/Downloads/micro-angular/blog/src/main/java/com/okta/developer/blog/service/UserService.java:37: error: cannot find symbol
    public UserService(UserRepository userRepository, AuthorityRepository authorityRepository) {
                                                      ^
  symbol:   class AuthorityRepository
  location: class UserService
/Users/mraible/Downloads/micro-angular/blog/src/main/java/com/okta/developer/blog/service/dto/AdminUserDTO.java:4: error: cannot find symbol
import com.okta.developer.blog.domain.Authority;
                                     ^
  symbol:   class Authority
  location: package com.okta.developer.blog.domain
/Users/mraible/Downloads/micro-angular/blog/src/main/java/com/okta/developer/blog/service/mapper/UserMapper.java:3: error: cannot find symbol
import com.okta.developer.blog.domain.Authority;
                                     ^
  symbol:   class Authority
  location: package com.okta.developer.blog.domain
/Users/mraible/Downloads/micro-angular/blog/src/main/java/com/okta/developer/blog/service/mapper/UserMapper.java:62: error: cannot find symbol
    private Set<Authority> authoritiesFromStrings(Set<String> authoritiesAsString) {
                ^
  symbol:   class Authority
  location: class UserMapper
warning: unknown enum constant When.MAYBE
  reason: class file for javax.annotation.meta.When not found
10 errors
2 warnings
```

**Review by mraible — APPROVED** [review]
Tested with the following steps:

```
jhipster jdl reactive-mf.jdl --monorepository --workspaces --client-framework angular
npm run java:docker:arm64
# start everything with docker compose up
npm run e2e -ws
```

**Comment by DanielFran** [comment]
@mshima can you fix snapshots?

File: generators/app/__snapshots__/generator.spec.ts.snap [hunk 1]
```diff
@@ -1575,7 +1575,7 @@ exports[`generator - app with microservice should match snapshot 1`] = `
   "gatewayServerPort": undefined,
   "gatlingTests": false,
   "generateAuthenticationApi": false,
-  "generateBuiltInAuthorityEntity": undefined,
+  "generateBuiltInAuthorityEntity": false,
```
Comment (mshima) [inline generators/app/__snapshots__/generator.spec.ts.snap:1578]:
```suggestion
  "generateBuiltInAuthorityEntity": undefined,
```

**Pull request merged**